honeybee_schema/cli.py

```diff
@@ -48,47 +49,55 @@ def update_model(model_json, version, output_file):
-
         # load the model dictionary from the json file and get the version of it
         with open(model_json) as json_file:
             model_dict = json.load(json_file)
         assert 'version' in model_dict, 'No version was found in the input model ' \
             'JSON. Update process cannot be run.'
+        print(f'Input model version: {model_dict["version"]}', file=sys.stderr)
         model_version = tuple(int(v) for v in model_dict['version'].split('.'))
 
-        # figure out which update modules need to run and import them
-        update_functions, update_vers = [], []
-        for mod, ver in zip(updaters, versions):
-            if ver < up_version and model_version < ver:
-                importlib.import_module(mod, 'honeybee_schema.updater')
-                update_vers.append(ver)
-                update_functions.append('version_{}_{}_{}'.format(*ver))
-        func_mapper = {}
-        base_state = 'getattr(subpackage.{0}, "{0}")'
-        for func in update_functions:
-            statement = base_state.format(func)
-            namespace = {'subpackage': honeybee_schema.updater}
-            func_mapper[func] = eval(statement, namespace)
-
-        # execute the update modules on the Model dictionary to update it
-        update_vers, update_functions = zip(*sorted(zip(update_vers, update_functions)))
-        statement = 'update_funct(model_dict)'
-        for up_func in update_functions:
-            namespace = {'update_funct': func_mapper[up_func], 'model_dict': model_dict}
-            model_dict = eval(statement, namespace)
+        if model_version >= up_version:
+            # no point for updating. Normally we should assert here but by writing
+            # the same file to a new file we make this command more flexible for cases
+            # that we don't know the version for the input model and we want to update
+            # it just in case.
+            mv = '.'.join(map(str, model_version))
+            uv = '.'.join(map(str, up_version))
+            print(
+                f'The input file has a higher version ({mv}) than the tagert version'
+                f' ({uv}).', file=sys.stderr
+            )
+            output_file.write(json.dumps(model_dict))
+            # let's consider exporting the same file as success
+            sys.exit(0)
+
+        # get functions with a higher version than model_version
+        updaters = []
+
+        for sub_module in pkgutil.walk_packages(updater.__path__):
+            if not sub_module.name.startswith('version_'):
+                continue
+            module = importlib.import_module(f'honeybee_schema.updater.{sub_module.name}')
+            for name, func in getmembers(module, isfunction):
+                if not name.startswith('version_'):
+                    continue
+                func_version = tuple(int(v) for v in name.split('_')[1:])
+                if model_version > func_version:
+                    continue
```

Comment:
This should be changed to the following:
```
if model_version > func_version or func_version > up_version:
    continue
```
We want to catch the case where a user inputs a version to upgrade their model that's less the the most recent version.

Reply:
Makes sense!